apps/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -221,22 +240,27 @@ export class ProjectChangeAnalyzer {
       }
     }
 
-    // Parallelize rush-project.json lookups
-    await Promise.all(
-      Array.from(changesByProject, async ([project, projectChanges]) => {
-        const filteredChanges: Map<string, IFileDiffStatus> = await this._filterProjectDataAsync(
-          project,
-          projectChanges,
-          repoRoot,
-          terminal
-        );
-        if (filteredChanges.size > 0) {
-          changedProjects.add(project);
-        }
-      })
-    );
-
-    return changedProjects;
+    if (forIncrementalBuild) {
+      const changedProjects: Set<RushConfigurationProject> = new Set();
+      await Async.forEachAsync(
+        changesByProject,
+        async ([project, projectChanges]) => {
+          const filteredChanges: Map<string, IFileDiffStatus> = await this._filterProjectDataAsync(
+            project,
+            projectChanges,
+            repoRoot,
+            terminal
+          );
+          if (filteredChanges.size > 0) {
+            changedProjects.add(project);
+          }
+        },
+        { concurrency: 50 }
```

Comment:
50 concurrent promises feels like it could easily hose a machine.  Looks like that call stack does disk operations including `RushProjectConfiguration.tryLoadForProjectAsync()`.  

Maybe reduce it to `3`?

Reply:
Reduced to 10